internal/fetchers/oci/manifest/manifest.go

```diff
@@ -0,0 +1,89 @@
+// Copyright The Enterprise Contract Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package manifest
+
+import (
+	"archive/tar"
+	"context"
+	"encoding/json"
+	"io"
+	"path"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	log "github.com/sirupsen/logrus"
+	"sigs.k8s.io/yaml"
+
+	"github.com/enterprise-contract/ec-cli/internal/fetchers/oci"
+)
+
+func ImageManifests(ctx context.Context, ref name.Reference, opts ...remote.Option) (map[string]json.RawMessage, error) {
+	img, err := oci.NewClient(ctx).Image(ref, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	content := mutate.Extract(img)
```

Comment:
I'm concerned about doing this for every image ec process. It seems unnecessary. Can we check for the presence of a certain label first? Even just any label that starts with `operatorframework.io/`.

Reply:
We check for presence of `operators.operatorframework.io.bundle.manifests.v1` in the rules, not saying that'll avoid doing unnecessary work, let me thinker a bit on this, I think I can make this semi-generic